modules/auxiliary/scanner/ssh/ssh_enum_git_keys.rb

```diff
@@ -0,0 +1,177 @@
+    )
+
+  end
+
+  # OPTPath will revert to pwd when set back to ""
+  def key_dir
+    datastore['KEY_DIR'] != `pwd`.strip ? datastore['KEY_DIR'] : ""
+  end
+
+  def key_file
+    datastore['KEY_FILE'] != `pwd`.strip ? datastore['KEY_FILE'] : ""
+  end
+
+  def has_passphrase?(file)
+    response = `ssh-keygen -y -P "" -f #{file} 2>&1`
+    return response.include? 'incorrect passphrase'
+  end
+
+  def read_keyfile(file)
+    if file.is_a? Array
+      keys = []
+      file.each do |dir_entry|
+        next unless ::File.readable? dir_entry
+
+        keys.concat(read_keyfile(dir_entry))
+      end
+      return keys
+    else
+      keyfile = ::File.open(file, "rb") { |f| f.read(f.stat.size) }
+    end
+    keys = []
+    this_key = []
+    in_key = false
+    keyfile.split("\n").each do |line|
+      in_key = true if (line =~ /^-----BEGIN ([RD]SA|OPENSSH) PRIVATE KEY-----/)
+      this_key << line if in_key
+      if (line =~ /^-----END ([RD]SA|OPENSSH) PRIVATE KEY-----/)
+        in_key = false
+        keys << file unless has_passphrase?(file)
+      end
+    end
+    if keys.empty?
+      print_error "SSH - No valid keys found"
+    end
+    return keys
+  end
+
+  def parse_user(output)
+    vprint_status("SSH Test: #{output}")
+    if (output =~ /You\'ve successfully authenticated/)
+      return output.match(/Hi (.*)\! You\'ve successfully authenticated/)[1]
+    elsif (output =~ /Welcome to GitLab, \@(.*)\!$/)
+      return output.match(/Welcome to GitLab, \@(.*)\!$/)[1]
+    end
+  end
+
+  def check_git_keys(queue)
+    threads = datastore['THREADS']
+    return {} if queue.blank?
+
+    threads = 1 if threads <= 0
+
+    results = {}
+    until queue.empty?
+      t = []
+      threads = 1 if threads <= 0
+
+      if queue.length < threads
+        threads = queue.length
+      end
+
+      begin
+        1.upto(threads) do
+          t << framework.threads.spawn("Module(#{refname})", false, queue.shift) do |file|
+            Thread.current.kill unless file
+
+            config_contents = "Host gitserver\n\tUser git\n\tHostname #{datastore['GITSERVER']}\n\tPreferredAuthentications publickey\n\tIdentityFile #{file}\n"
+
+            rand_file = Rex::Quickfile.new
+            rand_file.puts config_contents
+            rand_file.close
+
+            output = `ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -T -F #{rand_file.path} gitserver 2>&1`
+            if output.include? "\n"
+              output = output.split("\n")[-1]
+            end
+            user = parse_user(output)
+            if user
+              results[file] = user
+            end
+            rand_file.delete
+          end
+        end
+        t.map(&:join)
+      rescue ::Timeout::Error
+      ensure
+        t.each { |x| x.kill rescue nil }
+      end
+    end
+    return results
+  end
+
+  def test_keys
+    results = {}
```

Comment:
the content of results is never modified or accessed, can just get rid of it

Reply:
With the minor exception of https://github.com/rapid7/metasploit-framework/pull/12422/files#diff-25464e4273fbaa35a71e2fbdb2b237f4R150, which would be fine as `return {}`

Reply:
@acammack-r7 yup fair point, should have made that clear 😅

Reply:
Good catch. Updated